main.tf

```diff
@@ -388,6 +418,35 @@ resource "aws_route53_record" "atlantis" {
   }
 }
 
+################################################################################
+# EFS
+################################################################################
+
+resource "aws_efs_file_system" "this" {
+  count = var.enable_ephemeral_storage ? 0 : 1
+
+  creation_token = var.name
+}
+
+resource "aws_efs_mount_target" "this" {
+  # we coalescelist in order to specify the resource keys when we create the subnets using the VPC or they're specified for us.  This works around the for_each value depends on attributes which can't be determined until apply error
+  for_each = zipmap(coalescelist(var.private_subnets, var.private_subnet_ids), local.private_subnet_ids)
+
+  file_system_id  = aws_efs_file_system.this[0].id
```

Comment:
@MarkIannucci 
first of all, thank you! 🙏 

Unfortunately this is not working if I want to have `enable_ephemeral_storage=true`

```
╷
│ Error: Invalid index
│ 
│   on .terraform/modules/atlantis/main.tf line 438, in resource "aws_efs_mount_target" "this":
│  438:   file_system_id  = aws_efs_file_system.this[0].id
│     ├────────────────
│     │ aws_efs_file_system.this is empty tuple
│ 
│ The given key does not identify an element in this collection value.
╵
```

there is no condition for `var.enable_ephemeral_storage`

Reply:
no problem! thank you)